sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumer/EventHubConsumerLiveTests.cs

```diff
@@ -474,6 +540,308 @@ await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsum
+
+                    await using (var producer = client.CreateProducer(new EventHubProducerOptions { PartitionId = partition }))
+                    {
+                        // Sending some events beforehand so the partition has some information.
+
+                        for (var i = 0; i < 10; i++)
+                        {
+                            await producer.SendAsync(new EventData(new byte[1]));
+                        }
+
+                        // Store last enqueued time.
+
+                        var enqueuedTime = (await client.GetPartitionPropertiesAsync(partition)).LastEnqueuedTimeUtc;
+
+                        await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroupName, partition, EventPosition.FromEnqueuedTime(enqueuedTime)))
+                        {
+                            // Send a single event which is expected to go to the end of stream.
+                            // We are expecting to receive only this message.
+
+                            var stampEvent = new EventData(new byte[1]);
+                            stampEvent.Properties["stamp"] = Guid.NewGuid().ToString();
+
+                            await producer.SendAsync(stampEvent);
+
+                            // Receive and validate the events; because there is some non-determinism in the messaging flow, the
+                            // sent events may not be immediately available.  Allow for a small number of attempts to receive, in order
+                            // to account for availability delays.
+
+                            var receivedEvents = new List<EventData>();
+                            var index = 0;
+
+                            while ((receivedEvents.Count < 1) && (++index < ReceiveRetryLimit))
+                            {
+                                receivedEvents.AddRange(await consumer.ReceiveAsync(10, TimeSpan.FromMilliseconds(25)));
+                            }
+
+                            Assert.That(receivedEvents.Count, Is.EqualTo(1), "The number of received events should be 1.");
+                            Assert.That(receivedEvents.Single().IsEquivalentTo(stampEvent), Is.True, "The received event did not match the sent event.");
+
+                            // Next receive on this partition shouldn't return any more messages.
+
+                            Assert.That(await consumer.ReceiveAsync(10, TimeSpan.FromSeconds(2)), Is.Empty);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubConsumer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task ConsumerCanReceiveFromSequenceNumber(bool isInclusive)
```

Comment:
For context, Sequence Numbers may not be consecutive, so there's no true way to say "I don't want this one, but I do want the next."   That's why we're supporting the `IsInclusive` flag in this case.

Reply:
Fair enough.